aict-MERN/aict-site-w-react/src/components/Main/components/Posts/Posts.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react"
+import './posts.css'
+
+const Posts = () => {
+    const [posts, setPosts] = useState(null)
+
+    useEffect(() => {
+        fetch('http://localhost:3001/posts')
+            .then(res => res.json())
+            .then(data => {
+                const presentablePosts = data.posts.map((post) => {
+                    return <div key={post.id} className="post"><h2 className="post-author">{post.posted_by}</h2><p>{post.post}</p></div>
+                })
+                setPosts(presentablePosts)
+            })
+            .catch((err) => {
+                console.log(err)
+                setPosts(<div className="post"><p>Please configure database on your instance to create a post</p></div>)
```

Comment:
yes, you can do it like that, but its preferable that you dont pass around html in th ecode itself. same goes line 12.

the idiomatic way to do this would be to save a list of post objects on success and save an empty list on error and then transform the data into the corresponding html elements in the return on the last line.

Reply:
I did not know this. Im going to switch that around